numpy/distutils/fcompiler/environment.py

```diff
@@ -33,7 +33,7 @@ def __getattr__(self, name):
         try:
             conf_desc = self._conf_keys[name]
         except KeyError:
-            raise AttributeError(name)
+            raise AttributeError(name) from None
```

Comment:
While you're here, a better error message might be good, using
```
>>> object().missing
AttributeError: 'object' object has no attribute 'missing'
```
for inspiration

Reply:
You mean something like this:  raise AttributeError(dict_object['missing_attribute']) from None ?

Sorry for asking, but it helps me in my learning :)

Reply:
I mean make the text of the error look like my example above, instead of what it currently looks like,
```
AttributeError: missing
```

Reply:
Ok. Got it.
By the way in file numpy/ma/mrecords.py at Line: 201 "except (TypeError, KeyError):" isn't it better to separate except block in two except blocks?

Reply:
Only if the handling is different for the two errors. For that specific case, I wonder if `TypeError` can ever happen.